src/tools/kdb/set.cpp

```diff
@@ -35,11 +35,20 @@ int SetCommand::execute (Cmdline const & cl)
 	Key k = cl.createKey (0);
 	std::string name = k.getName ();
 	Key parentKey = cl.getParentKey (k);
+	Key cascadingParentKey = parentKey.dup ();
+	cascadingParentKey.setNamespace (kdb::ElektraNamespace::CASCADING);
 
 	// do not resume on any get errors
 	// otherwise the user might break
 	// the config
-	kdb.get (conf, parentKey);
+	if (cl.force)
+	{
+		kdb.get (conf, parentKey);
+	}
+	else
+	{
+		kdb.get (conf, cascadingParentKey);
```

Comment:
Maybe add the declaration of `cascadingParentKey` into this else block, as it isn't used anywhere else.
Also please add a comment stating why we use the `parentKey` as-is if `cl.force` is true, and why the cascading key if it is not.

Reply:
Makes sense, added a comment and moved it into the if block